pkg/cloud/services/network/gateways_test.go

```diff
@@ -149,3 +150,117 @@ func TestReconcileInternetGateways(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteInternetGateways(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	testCases := []struct {
+		name    string
+		input   *infrav1.NetworkSpec
+		expect  func(m *mock_ec2iface.MockEC2APIMockRecorder)
+		wantErr bool
+	}{
+		{
+			name: "Should return without error if vpc is unmanaged",
+			input: &infrav1.NetworkSpec{
+				VPC: infrav1.VPCSpec{
+					ID: "vpc-gateways",
+				},
+			},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {},
+		},
+		{
+			name: "Should return without error if internet gateway is not found",
+			input: &infrav1.NetworkSpec{
+				VPC: infrav1.VPCSpec{
+					ID: "vpc-gateways",
+					Tags: infrav1.Tags{
+						infrav1.ClusterTagKey("test-cluster"): "owned",
+					},
+				},
+			},
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+				m.DescribeInternetGateways(gomock.Eq(&ec2.DescribeInternetGatewaysInput{
+					Filters: []*ec2.Filter{
+						{
+							Name:   aws.String("attachment.vpc-id"),
+							Values: aws.StringSlice([]string{"vpc-gateways"}),
+						},
+					},
+				})).Return(&ec2.DescribeInternetGatewaysOutput{}, nil)
+			},
+		},
+		{
+			name: "Should return without error if internet gateway is not found",
```

Comment:
I think you are testing a successful deletion of IGW. Could you change the test name to reflect it?